sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -1601,7 +1601,7 @@ private void showDisabledToast() {
     synchronized void storeDownloadRequestId(DownloadManager downloadManager, DownloadTask task, long downloadId, long enqueueTime) {
 
         /* Check for if state changed and task not canceled in time. */
-        if (mDownloadTask == task) {
+        if (mDownloadTask == task && mReleaseDetails != null) {
```

Comment:
We are preventing a crash but the root cause is unknown, how can we have a download task still pending but losing the release details object? We need to understand the root cause otherwise we will have downloads not working and we won't know why.